torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -394,6 +399,34 @@ TensorView* TensorView::reorder(const std::unordered_map<int, int>& old2new_) {
   return this;
 }
 
+TensorView* TensorView::swizzle(
+    SwizzleType type,
+    const std::vector<int>& axes) {
+  swizzle_type_ = type;
+
+  // Clear previously set swizzle axes if any
+  if (axes_to_swizzle_.size()) {
+    axes_to_swizzle_.clear();
+  }
+
+  if (swizzle_type_ == SwizzleType::Transpose) {
+    TORCH_CHECK(axes.size() == 2);
```

Comment:
Should we also check that we don't get the same axis twice?

Reply:
Yes. I also added more descriptive error messages.